src/map/script_map.cpp

```diff
@@ -1056,6 +1056,69 @@ static int CclGetIsGameHoster(lua_State *l)
 	return 1;
 }
 
+/**
+** <b>Description</b>
+**
+** Set basic map caracteristics.
+**
+**  @param l  Lua state.
+**
+** Example:
+**
+** <div class="example"><code><strong>PresentMap</strong>("Map description", 1, 128, 128, 17)</code></div>
+*/
+static int CclPresentMap(lua_State *l)
+{
+	LuaCheckArgs_min(l, 5);
+
+	Map.Info.Description = LuaToString(l, 1);
+	// Number of players in LuaToNumber(l, 3); // Not used yet.
+	Map.Info.MapWidth = LuaToNumber(l, 3);
+	Map.Info.MapHeight = LuaToNumber(l, 4);
+	Map.Info.MapUID = LuaToNumber(l, 5);
+	
+	if(LuaGetArgsNum(l) >= 6) {
+		const std::string_view highgrounds = LuaToString(l, 6);
+		if (highgrounds == "highgrounds-enabled") {
+			Map.Info.EnableHighgrounds();
+		}
+	}
+
+	return 0;
+}
+
+static int CclMapEnableHighgrounds(lua_State *l)
+{
+	Map.Info.EnableHighgrounds(LuaGetArgsNum(l) >= 1 ? LuaToBoolean(l, 1) : true);
+	
+	return 0;
+}
+
+static int CclIsHighgroundsEnabled(lua_State *l)
+{
+	lua_pushboolean(l, Map.Info.IsHighgroundsEnabled());
+	return 1;
+}
+
+/**
+** <b>Description</b>
+**
+** Define the lua file that will build the map
+**
+**  @param l  Lua state.
+**
+** Example:
+**
+** <div class="example"><code>-- Load map setup from file
+**		<strong>DefineMapSetup</strong>("Setup.sms")</code></div>
+*/
+static int CclDefineMapSetup(lua_State *l)
+{
+	LuaCheckArgs(l, 1);
+	Map.Info.Filename = LuaToString(l, 1);
+
+	return 1;
```

Comment:
```suggestion
	return 0;
```
Nothing is returned in Lua...

Reply:
Not sure about this returns.
You mean we should return `1` if we pass something back to lua and `0` otherwise?

Reply:
The return value correspond to the number of return value (as tuple from lua stack).